### PR TITLE
feat: change list stages

### LINKED ISSUE #50: Changing list stages
Currently, lists have 3 default stages: To Do, In Progress, and Completed. We need a way to customize these stages and add or delete them.

### PR DESCRIPTION
**Pull request by Us3r-gitHub**
Closes #50

### CODE REVIEW

File: packages/backend/src/app/common/lists/lists.controller.ts [hunk 1]
```diff
@@ -26,8 +27,8 @@ export class ListsController {
     constructor(private readonly listsService: ListsService) {}
 
     @Get()
-    findAll(): Promise<ListFindAllResult> {
-        return this.listsService.findAll();
+    findAll(@Query("workspaceId") spaceId: number): Promise<ListFindAllResult> {
```
Comment (MoShizzle) [inline packages/backend/src/app/common/lists/lists.controller.ts:30]: This should be workspaceId no?

Comment (Us3r-gitHub) [reply packages/backend/src/app/common/lists/lists.controller.ts:30]: Refer to [this query parameter](https://github.com/tillywork/tillywork/blob/main/packages/frontend/src/composables/services/useListsService.ts#L22) using the `workspaceId` key

Meanwhile when defining [the entity](https://github.com/tillywork/tillywork/blob/main/packages/backend/src/app/common/lists/list.entity.ts#L37), use the `spaceId` key

Comment (Us3r-gitHub) [reply packages/backend/src/app/common/lists/lists.controller.ts:30]: > This should be spaceId no?

I just found out that workspace and space are two different things
I will make a minor change to the front end

File: packages/frontend/src/components/common/tabs/SettingTab/SettingTabListStages.vue [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  </v-card>
+</template>
+
+<script setup lang="ts">
+import { DIALOGS } from '../../dialogs/types';
+import { useDialogStore } from '@/stores/dialog';
+import { useListStagesService } from '@/composables/services/useListStagesService';
+import { type ListStage } from '@/components/project-management/lists/types';
+
+const listId = ref(3); // TODO Implement Get Current/Selected ListId
```
Comment (Us3r-gitHub) [inline packages/frontend/src/components/common/tabs/SettingTab/SettingTabListStages.vue:75]: I've already implemented
If access `Settings` dialog from navigation menu item, just pass `list` and extract the id

However, I'm still wondering
How can we retrieve list when accessing `Setting` dialog from this button?
image

Any suggestion?

Comment (MoShizzle) [reply packages/frontend/src/components/common/tabs/SettingTab/SettingTabListStages.vue:75]: For now, we can show a v-autocomplete that shows lists grouped by space in the settings dialog, so the user can select which list to edit. The options in the v-autocomplete would look something like this, with only the lists being selectable:
`Space 1
  List 1
  List 2
Space 2
  List 3
  List 4`

Comment (Us3r-gitHub) [reply packages/frontend/src/components/common/tabs/SettingTab/SettingTabListStages.vue:75]: So, would it look like onboarding?

The idea is to select the list before rendering the 'List Stages' tab in the Settings dialog

**Comment by MoShizzle** [comment]
There seems to be a bug when deleting list stages: If the stage has cards, and I choose which stages I want the cards moved to, the cards are moved, the list stage is deleted, but the list group still shows in the view.

**Comment by Us3r-gitHub** [comment]
> There seems to be a bug when deleting list stages: If the stage has cards, and I choose which stages I want the cards moved to, the cards are moved, the list stage is deleted, but the list group still shows in the view. This only happens when the deleted stage has cards assigned to it.

Already fixed in this db6d5e611d93e43d418be488821c2322a610710e

**Review by MoShizzle — APPROVED** [review]

**Pull request merged**